### DISCUSSION TITLE
Confetti animation on new ViewController

### DISCUSSION THREAD

**Discussion by juliusbackes** [simibac/ConfettiSwiftUI#30] [General]
```
import UIKit
import SwiftUI
import ConfettiSwiftUI

class happyResultVC: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
    }
 

    @State private var counter: Int = 0
   func confetti() {
       counter += 1
    ConfettiCannon(counter: $counter)
    }
    
  
   if viewController.viewIfLoaded?.window != nil {
        confetti()
    }
}
```
![Simulator Screen Shot - iPhone 14 Pro - 2022-12-30 at 18 00 35](https://user-images.githubusercontent.com/115353924/210097253-aeddfd00-105c-43c3-bfc4-8fe606af6b3f.png)
![Simulator Screen Shot - iPhone 14 Pro - 2022-12-30 at 18 00 42](https://user-images.githubusercontent.com/115353924/210097258-2bcfc30b-887a-4246-a65e-2b7bfae46bff.png)

So I'm trying that if the "happy number"-Result Screen (Picture 2)opens that the confetti animation shows up. I'm using UIKit in my whole project. Is that possible? Thank you in advance!

**Comment by simibac** [comment]
I dont have much experience with UIKit. I would assume that you need additional logic to use SwiftUI inside UIKit. Maybe this tutorial can help you: https://developer.apple.com/tutorials/swiftui/interfacing-with-uikit

**Reply by juliusbackes** [reply]
Okay thank you. But is it possible that you don't use a button. Like if the View appears then the animation starts automatically. Thank you!
